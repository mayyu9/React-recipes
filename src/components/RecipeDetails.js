import React from 'react';
import classNames from 'classnames';

const RecipeDetails = (props) =>{
  if(! props.currentRecipes){
    return (
      <p
      className={ classNames('h3 p2 bg-white italic center', props.className)}
        style={props.style}
      >
      Please the recipe to check all other deatils.
      </p>
    )
  }
  return(
    <div
      className={classNames('p2 bg-white', props.className)}
      style={props.style}
    >
      <h2 className="h2">
        {props.currentRecipes.name}
      </h2>
      <img className="fit" alt={props.currentRecipes.name} src={props.currentRecipes.image} style={{maxWidth: '100%'}} />
      <div>
        <span>category: {props.currentRecipes.category} </span>
        <span> Calories: {props.currentRecipes.calories}</span>
      </div>
      <h3>Ingredients</h3>
      <ul>
        {props.currentRecipes.ingredients.map(ingredient => {
          return(
          <li key={ingredient}>
            {ingredient}
          </li>
        );
        })
      }
      </ul>
      <h3>steps</h3>
      <ul>
        {props.currentRecipes.steps.map( (step,index) => {
          return(
          <li key={index+1}>
            {step}
          </li>
        );
        })
      }
      </ul>
    </div>
  )
};

export default RecipeDetails;
