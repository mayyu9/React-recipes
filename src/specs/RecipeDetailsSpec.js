import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import RecipeDetails from '../components/RecipeDetails';

// const testRecipe = {
//   id: 1,
//   name: 'test Recipe',
//   category: 'test category',
//   ingredient: ['intg 1', 'intg 2', 'intg 3'],
//   steps: ['step 1', 'step 2', 'step 3'],
// };

// const testRecipeIntg = {
//   id: 1,
//   name: 'test Recipe',
//   category: 'test category',
//   steps: ['step 1', 'step 2', 'step 3'],
// };
//
// const testRecipeStep = {
//   id: 1,
//   name: 'test Recipe',
//   category: 'test category',
//   ingredient: ['intg 1', 'intg 2', 'intg 3'],
// };

describe('<RecipeDetails />', () => {
  let testRecipe;
  beforeEach(() => {
    testRecipe = {
      id: 1,
      name: 'test Recipe',
      category: 'test category',
      ingredient: ['intg 1', 'intg 2', 'intg 3'],
      steps: ['step 1', 'step 2', 'step 3'],
    };
  });

  test('should render RecipeDetails component, with zero state', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetails />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render RecipeDetails component, with recipe details', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetails currentRecipes={testRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render RecipeDetails component, with recipe className', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetails currentRecipes={testRecipe} className="test-className" />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render RecipeDetails component, recipe without ingredients', () => {
    delete testRecipe.ingredient;
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetails currentRecipes={testRecipe} className="test-intg" />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render RecipeDetails component, recipe without steps', () => {
    delete testRecipe.steps;
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetails currentRecipes={testRecipe} className="test-step" />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
