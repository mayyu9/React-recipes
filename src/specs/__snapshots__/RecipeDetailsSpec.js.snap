// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetails /> should render RecipeDetails component, recipe without ingredients 1`] = `
<div
  className="p2 bg-white test-intg"
>
  <h2
    className="h2"
  >
    test Recipe
  </h2>
  <img
    alt="test Recipe"
    className="fit"
    style={
      Object {
        "maxWidth": "100% ",
      }
    }
  />
  <div>
    <span>
      category: 
      test category
       
    </span>
    <span>
       Calories: 
    </span>
  </div>
  <h3>
    Ingredients
  </h3>
  <h3>
    steps
  </h3>
  <ul>
    <li>
      step 1
    </li>
    <li>
      step 2
    </li>
    <li>
      step 3
    </li>
  </ul>
</div>
`;

exports[`<RecipeDetails /> should render RecipeDetails component, recipe without steps 1`] = `
<div
  className="p2 bg-white test-step"
>
  <h2
    className="h2"
  >
    test Recipe
  </h2>
  <img
    alt="test Recipe"
    className="fit"
    style={
      Object {
        "maxWidth": "100% ",
      }
    }
  />
  <div>
    <span>
      category: 
      test category
       
    </span>
    <span>
       Calories: 
    </span>
  </div>
  <h3>
    Ingredients
  </h3>
  <h3>
    steps
  </h3>
</div>
`;

exports[`<RecipeDetails /> should render RecipeDetails component, with recipe className 1`] = `
<div
  className="p2 bg-white test-className"
>
  <h2
    className="h2"
  >
    test Recipe
  </h2>
  <img
    alt="test Recipe"
    className="fit"
    style={
      Object {
        "maxWidth": "100% ",
      }
    }
  />
  <div>
    <span>
      category: 
      test category
       
    </span>
    <span>
       Calories: 
    </span>
  </div>
  <h3>
    Ingredients
  </h3>
  <h3>
    steps
  </h3>
  <ul>
    <li>
      step 1
    </li>
    <li>
      step 2
    </li>
    <li>
      step 3
    </li>
  </ul>
</div>
`;

exports[`<RecipeDetails /> should render RecipeDetails component, with recipe details 1`] = `
<div
  className="p2 bg-white"
>
  <h2
    className="h2"
  >
    test Recipe
  </h2>
  <img
    alt="test Recipe"
    className="fit"
    style={
      Object {
        "maxWidth": "100% ",
      }
    }
  />
  <div>
    <span>
      category: 
      test category
       
    </span>
    <span>
       Calories: 
    </span>
  </div>
  <h3>
    Ingredients
  </h3>
  <h3>
    steps
  </h3>
  <ul>
    <li>
      step 1
    </li>
    <li>
      step 2
    </li>
    <li>
      step 3
    </li>
  </ul>
</div>
`;

exports[`<RecipeDetails /> should render RecipeDetails component, with zero state 1`] = `
<p
  className="h3 p2 bg-white italic center"
>
  Please the recipe to check all other deatils.
</p>
`;
